### PR TITLE
Allow TextProgressBar to print to different handlers

### LINKED ISSUE #1785: Support the progress bar to another stream handler
### What is the expected enhancement?
ProgressBar is a good visualization tool when you have tons of jobs running in the background. 

Now, the progress bar will be printed in `sys.stdout`, I wonder that is it possible to add an argument to `TextProgressBar` to print the progress on `sys.stderr` or the other handler? 
Then, users can control when the progress bar is displayed. (I know that users can always write the wrapper to wrap `sys.stdout` to the logger but it seems to be a workaround.)

Thanks.

### PR DESCRIPTION
**Pull request by maddy-tod**
### Summary
Fixes #1785 

Added in the parameter `output_handler` to the `TextProgressBar` to allow the user to send the progress bar to a different output stream. This can be specified by providing the handler itself, e.g. `sys.stderr`. This means that if a user wanted to create their own output stream, as long as it had a write method, it could still be used with this drawer.

### CODE REVIEW

File: qiskit/tools/events/progressbar.py [hunk 1]
```diff
@@ -109,12 +109,26 @@ def finished(self):
...
     """
 
-    def __init__(self):
+    def __init__(self, output_handler=None):
         super().__init__()
         self._init_subscriber()
 
+        self.output_handler = sys.stdout
+
+        if isinstance(output_handler, str):
```
Comment (diego-plan9) [inline qiskit/tools/events/progressbar.py]: Would it make sense to just accept a handler directly as `output_handler`, that might have also the nice side effect of just being able to set `sys.stdout` as the default? This way we would not need to involve a "magic string" in the process and might be clearer for end users.

Comment (maddy-tod) [reply qiskit/tools/events/progressbar.py]: I did think this, but also I felt that allowing the user to enter a string might be more helpful to less experienced python users. Currently, a user can send a handler directly if they wish to, but I can make this the only option if that would be preferred.

Comment (nonhermitian) [reply qiskit/tools/events/progressbar.py]: I would agree that the strings are unnecessary. The docstrings can layout the options, and it is unlikely that an inexperienced Python user would actually need such a feature.

Comment (maddy-tod) [reply qiskit/tools/events/progressbar.py]: That is a good point, I will update it now.

**Pull request merged**